src/Resources/views/search/index.html.twig

```diff
@@ -1,6 +1,11 @@
 {# @var items \Sylius\Component\Core\Model\ProductInterface[] #}
 {% extends '@SyliusShop/layout.html.twig' %}
 
+{% block metatags %}
```

Comment:
I am more fan of adding the canonical link as a response header. Then it's also easier for people to manipulate/remove it later on

Reply:
You can see how it's done in Symfony here: `\Sylius\Bundle\ResourceBundle\Controller\ControllerTrait::addLink`

Reply:
Should I remove the Twig extension then?

Reply:
I think so, yes

Reply:
Is using the link Twig function okay instead of writing an event subscriber?

Reply:
Not sure what you mean?

Reply:
I mean using the `link` Twig function ([ref](https://github.com/symfony/symfony/blob/468c22b960afea6a486231d635953e4cc5ef0bf7/src/Symfony/Bridge/Twig/Extension/WebLinkExtension.php#L35)) to add the canonical link header.

Reply:
That seems counter intuitive :D I would rather go for the dedicated event subscriber :)

Reply:
That was my first thought, too, but it's more code to write and maintain, and we know we'll always have Twig, so what's the added value of writing an event subscriber?

Reply:
It could be like this:
![image](https://github.com/user-attachments/assets/979055b4-7788-47e9-89ad-ce2ed79f1397)
We can always replace it with an event subscriber later if it proves necessary.

Reply:
I will also accept an answer like "I prefer it this way" without other arguments. 😄

Reply:
That indeed looks nice, but still 

> so what's the added value of writing an event subscriber

... that users of the plugin doesn't have to think about it

Reply:
Then I have a few more questions about the subscriber implementation. A simple subscriber that will add the canonical link to all responses. Is that okay? I suppose we want it just for the search/taxon index responses. In that case, I see 3 options:
1. The good old checking of the route name in the subscriber.
2. Use the SearchRequestCreated or SearchResponseParametersCreated events from SearchAction and add the link to the request attributes.
2. Dispatch a dedicated event in SearchAction with the response object and add the header directly.